alchemy/src/vercel/storage.ts

```diff
@@ -0,0 +1,184 @@
+import type { Context } from "../context.ts";
+import { Resource } from "../resource.ts";
+import type { Secret } from "../secret.ts";
+import { createVercelApi, VercelApi } from "./api";
+
+type StorageType = "blob";
+
+/**
+ * Properties for creating or updating a Storage
+ */
+export interface StorageProps {
+  /**
+   * The desired name for the storage
+   */
+  name: string;
+
+  /**
+   * The region where the storage will be deployed
+   */
+  region: string;
+
+  /**
+   * The team ID that the storage belongs to
+   */
+  teamId: string;
+
+  /**
+   * The type of storage
+   */
+  type: StorageType;
+}
+
+/**
+ * Output returned after Storage creation/update
+ */
+export interface Storage extends Resource<"vercel::Storage">, StorageProps {
+  /**
+   * The ID of the storage
+   */
+  id: string;
+
+  /**
+   * The storage store information
+   */
+  store: {
```

Comment:
Good to pull this out into its own interface since it's a nested object

Reply:
Or should we just flatten these properties into the `Storage` interface?

Reply:
i ended up merging them in the `Storage` interface, however, it does feel "messy", because we are merging the API response to it and some properties can overlap... but looking at other resources, they are all doing that (aka merging back to the main interface), so i'm just following it too